sdk/src/driftClient.ts

```diff
@@ -6554,21 +6650,31 @@ export class DriftClient {
 			allIx.push(instructions);
 		}
 
-		txVersion = txVersion ?? this.txVersion;
+		const latestBlockHashAndContext = await this.connection.getLatestBlockhashAndContext({
+			commitment: this.opts.preflightCommitment,
+		});
+
+		// # Create and return Transaction
 		if (txVersion === 'legacy') {
-			return new Transaction().add(...allIx);
+			if (forceVersionedTransaction) {
+				const message = new TransactionMessage({
+					payerKey: this.provider.wallet.publicKey,
+					recentBlockhash: latestBlockHashAndContext.value.blockhash,
+					instructions: allIx,
+				}).compileToLegacyMessage();
```

Comment:
this is the path where it's a legacy transaction but we're forcing it into a VersionedTransaction object because you can't simulate with the legacy transaction objects .. but it's still a legacy one so won't work with lookupTables an stuff right? 

You can see just below we use the `compileToV0Message` if it's actually a versioned transaction